Fixed the bug, a null able list does not work

sikandar opened the pull request:
If there is a nullable list with values, the implementation should check it instead of wrapping the nullable list (even if it has values) into another newly created list by the Criteria class.

### Existing
Code: `if (!filter.ids.isNullOrEmpty()) criteria.and("id").`in`(filter.ids) `
Output: `{"id" : { "$in" : [["667e8b8af76f17213e4d4280"]]}}` **Wrong**

Code: `if (!filter.ids.isNullOrEmpty()) criteria.and("id").`in`(filter.ids!!)` **Required !!**
Output:` {"id" :{ "$in" : ["667e8b8af76f17213e4d4280"]}}` **Correct**

### After Fix
Code: `if (!filter.ids.isNullOrEmpty()) criteria.and("id").`in`(filter.ids) ` **With and Without !!**
Output: `{"id" : { "$in" : ["667e8b8af76f17213e4d4280"]}}` **Correct**

<!--

Thank you for proposing a pull request. This template will guide you through the essential steps necessary for a pull request.
Make sure that:

-->

- [x] You have read the [Spring Data contribution guidelines](https://github.com/spring-projects/spring-data-build/blob/master/CONTRIBUTING.adoc).
- [x] You use the code formatters provided [here](https://github.com/spring-projects/spring-data-build/tree/master/etc/ide) and have them applied to your changes. Don’t submit any formatting related changes.
- [x] You submit test cases (unit or integration tests) that back your changes.
- [x] You added yourself as author in the headers of the classes you touched. Amend the date range in the Apache license header if needed. For new types, add the license header (copy from another file and set the current year only).

sikandar committed:
Fixed the bug, a null able list does not work

If there is a nullable list with values, the implementation should check it instead of wrapping the nullable list (even if it has values) into another newly created list by the Criteria class.

if (!filter.ids.isNullOrEmpty()) criteria.and("id").`in`(filter.ids)
-> {"id" : { "$in" : [["667e8b8af76f17213e4d4280"]]}}

if (!filter.ids.isNullOrEmpty()) criteria.and("id").`in`(filter.ids!!)
-> {"id" :{ "$in" : ["667e8b8af76f17213e4d4280"]}}

pivotal-cla commented:
@sikandar Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-data-mongodb&pullRequestId=4734)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-projects/spring-data-mongodb&pullRequestId=4734) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

pivotal-cla commented:
@sikandar Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-data-mongodb&pullRequestId=4734)!

sikandar committed:
Fixed the criteria in method

If there is a nullable list with values, the implementation should check it instead of wrapping the nullable list (even if it has values) into another newly created list by the Criteria class.

Existing
Code: if (!filter.ids.isNullOrEmpty()) criteria.and("id").in(filter.ids) 
Output: {"id" : { "$in" : [["667e8b8af76f17213e4d4280"]]}} Wrong

Code: if (!filter.ids.isNullOrEmpty()) criteria.and("id").in(filter.ids!!) Required !!
Output: {"id" :{ "$in" : ["667e8b8af76f17213e4d4280"]}} Correct

After Fix
Code: if (!filter.ids.isNullOrEmpty()) criteria.and("id").in(filter.ids)  With and Without !!
Output: {"id" : { "$in" : ["667e8b8af76f17213e4d4280"]}} Correct

sikandar committed:
Fixed the bug, a null able list does not work

If there is a nullable list with values, the implementation should check it instead of wrapping the nullable list (even if it has values) into another newly created list by the Criteria class.

Existing
Code: if (!filter.ids.isNullOrEmpty()) criteria.and("id").in(filter.ids) 
Output: {"id" : { "$in" : [["667e8b8af76f17213e4d4280"]]}} Wrong

Code: if (!filter.ids.isNullOrEmpty()) criteria.and("id").in(filter.ids!!) Required !!
Output: {"id" :{ "$in" : ["667e8b8af76f17213e4d4280"]}} Correct

After Fix
Code: if (!filter.ids.isNullOrEmpty()) criteria.and("id").in(filter.ids)  With and Without !!
Output: {"id" : { "$in" : ["667e8b8af76f17213e4d4280"]}} Correct

 You have read the Spring Data contribution guidelines.
 You use the code formatters provided here and have them applied to your changes. Don’t submit any formatting related changes.
 You submit test cases (unit or integration tests) that back your changes.
 You added yourself as author in the headers of the classes you touched. Amend the date range in the Apache license header if needed. For new types, add the license header (copy from another file and set the current year only).

christophstrobl commented:
Thank you for getting in touch. Please explain why you think this is necessary and provide throughout testcases for the proposed change since there's a method that accepts a `Collection`. 

As `{"$in" : [["n1", "n2"]]}` matches differently from `{"$in" : ["n1", "n2"]}` I think revisiting the arrangement makes sense.

sikandar commented:
> Thank you for getting in touch. Please explain why you think this is necessary and provide throughout testcases for the proposed change since there's a method that accepts a `Collection`.
> 
> As `{"$in" : [["n1", "n2"]]}` matches differently from `{"$in" : ["n1", "n2"]}` I think revisiting the arrangement makes sense.

This solution will be able to work with Variable Arguments
